### PR TITLE
feat(#96): replace scheduleSuppressedUntil with scheduleLastStoppedAt + foreground catch-up

### LINKED ISSUE #96: Schedule doesn't start after configuration (suppressedUntil race)
## Bug

Configured a profile schedule at 20:52 to start at 20:54. Pre-activation reminder fired at ~20:53 and vanished just before 20:54, but the schedule **never started blocking**.

## What We Know

1. **The DeviceActivity extension DID fire.** The pre-activation reminder vanishing proves `ScheduleTimerActivity.start()` executed — it cancels reminders on lines 37-43 *before* any guard checks.

2. **`saveToProfile()` sets `scheduleSuppressedUntil` to the immediate next occurrence.** In `TriggerConfigurationModel.swift:106`:
   ```swift
   profile.scheduleSuppressedUntil = profile.startSchedule?.nextScheduledStartTime(after: Date())
   ```

3. **`nextScheduledStartTime(after:)` returns TODAY's occurrence when it hasn't passed yet.** At 20:52:05, calling `nextScheduledStartTime(after: 20:52:05)` for a 20:54 schedule returns **today at 20:54:00** — because 20:52 < 20:54 makes today the candidate (`ProfileScheduleTime.swift:52`).

4. **The suppress guard blocks the start.** `ScheduleTimerActivity.swift:71`:
   ```swift
   if let suppressedUntil = profile.scheduleSuppressedUntil, Date() < suppressedUntil {
       return  // ← schedule blocked
   }
   ```

5. **The `isOldEnough` guard passes.** Schedule saved at 20:52:05, fires at ~20:54 = ~115 seconds > 60 second threshold. Not the culprit.

## What We Suspect

**Root cause:** `scheduleSuppressedUntil` is set to 20:54:00 at save time (20:52:05). When `intervalDidStart` fires at ~20:54:00, the check `Date() < 20:54:00` either:
- Returns `true` because DeviceActivity fired slightly early (20:53:59.xxx), OR
- Returns `true` because of Date precision/rounding

Either way, the **first scheduled occurrence after configuration is suppressed**. This is a bug in the suppress feature (#88 / PR #92) — the suppress was intended for "after manual stop" but `saveToProfile()` also applies it on initial configuration and every schedule edit.

**The fix is likely:** Don't set `scheduleSuppressedUntil` in `saveToProfile()`, OR set it to a time that has already passed (e.g., `Date.distantPast`), OR set it to the *second* next occurrence instead of the immediate next one.

## What We Want to Know

1. **Extension logs** — can you reproduce and check the device monitor extension logs? They should show either `"suppressed until ..."` or a different guard message, confirming which guard blocked the start.

2. **Does this reproduce with longer lead times?** E.g., configure at 3:00 PM for a 9:00 PM start — does the suppress check pass because `Date()` at 9:00 PM is clearly `>= suppressedUntil` (9:00 PM)?

3. **Was the `saveToProfile()` suppress intentional?** The manual-stop suppress in `BlockedProfileSessions.swift:104` makes sense (prevent restart in same window). But why suppress on initial configuration?

## Log Evidence

Full log: `docs/plans/foqos-current.log`

Key lines:
```
[20:52:05] Scheduled restrictions from hour: 20 minute: 54 to hour: 20 minute: 53 daily
[20:52:05] Scheduled 1 pre-activation reminder(s) for TikTok
           ← reminder fires at ~20:53, vanishes before 20:54
           ← 20:54 passes, no session starts
[20:54:59] Scheduled restrictions from hour: 20 minute: 54 to hour: 20 minute: 53 daily
           ← re-registered AFTER start time, won't fire until tomorrow
```

## Secondary Concern

The schedule is also re-registered on every foreground event (via `PreActivationReminderScheduler.rescheduleAllReminders()` → `scheduleTimerActivity()`). The cancel-and-re-register at 20:53:20 (40 seconds before the 20:54 trigger) is probably fine, but the re-registration at 20:54:59 (after the start time passed) means the schedule won't fire again until tomorrow. This amplifies the primary bug — even if the first trigger was suppressed, a retry on the next foreground won't help because the window has passed.

### PR DESCRIPTION
**Pull request by mnbf9rca**
## Summary
- Replace derived future timestamp `scheduleSuppressedUntil` with recorded fact `scheduleLastStoppedAt` to fix race conditions (#96)
- Add `activeWindowStart` helper that correctly computes schedule windows for same-day, overnight, and start-only schedules
- Add `shouldBeActiveNow` as single source of truth for "should this schedule be running?" — consolidates window detection, day check, age check, and suppression
- Add foreground catch-up mechanism: when the app returns to foreground, missed schedule starts are detected and started immediately
- Fix overnight schedule bug: suppression now correctly handles windows spanning midnight (e.g., 22:00-06:00)
- Simplify `ScheduleTimerActivity.start()` V2 path to delegate entirely to `shouldBeActiveNow`
- Remove writers' suppression computation — writers now just record `scheduleLastStoppedAt = Date()`
- 30 new tests across 4 test files, all using pinned reference dates for determinism

### Original: Race conditions with derived timestamps
`scheduleSuppressedUntil` was a derived future timestamp. Every writer had to compute "when should suppression expire?" creating 5-second tolerance hacks, complex `saveToProfile` logic, and stale-snapshot bugs.

### Additional: Overnight schedule bug
`shouldSuppressStart` used `scheduledStartTime(on: now)` which always returns today's start time. For an overnight schedule (22:00-06:00), at 03:00 AM the "today start" is today at 22:00 — but the actual window started yesterday at 22:00. This caused incorrect suppression decisions.

### Additional: Missed schedule starts on foreground
When the app returns to foreground after a schedule's start time has passed, DeviceActivity may not re-fire `intervalDidStart`. Scheduled sessions that should be running are silently missed.

### 1. Record facts, decide at read time
**Writers** record `scheduleLastStoppedAt = Date()`. No computation needed.
**Readers** evaluate at fire time using `shouldBeActiveNow`.

### 2. Correct window detection with `activeWindowStart`
Returns the start Date of the currently active schedule window:
- **Same-day** (10:00-17:00): returns today's 10:00 if between 10:00 and 17:00
- **Overnight** (22:00-06:00): returns today's 22:00 if after 22:00, or yesterday's 22:00 if before 06:00
- **Start-only** (no stop): returns today's start if past start time

### 3. Consolidated `shouldBeActiveNow`
Single source of truth combining:
1. Window detection (via `activeWindowStart`)
2. Day-of-week check (uses window start date, not current date — handles overnight correctly)
3. Age check (prevents immediate fire on save)
4. Suppression check (compares window start against `scheduleLastStoppedAt`)

### 4. Foreground catch-up
`catchUpMissedScheduleStarts()` runs on both `.onAppear` and `.onChange(of: scenePhase)`. For each V2 profile where `shouldBeActiveNow` returns true but no session is active, it calls `ScheduleTimerActivity().start()` directly. Deduplication is safe — `start()` checks for existing active sessions.

## Key scenarios

| Scenario | Comparison | Result |
|----------|-----------|--------|
| Manual stop at 10:05, DA re-fires at 10:06 | window 10:00 ≤ 10:05 | **Suppress** |
| Stopped yesterday at 10:05, fires today at 10:00 | window today 10:00 > yesterday 10:05 | **Fire** |
| Overnight 22:00-06:00, at 03:00, stopped 22:30 yesterday | window yesterday 22:00 ≤ yesterday 22:30 | **Suppress** |
| Overnight 22:00-06:00, at 03:00, not stopped | window yesterday 22:00, no stop | **Fire** |
| App foregrounds at 14:00, schedule was 10:00, no session | shouldBeActiveNow = true | **Catch up** |

## Files Changed (16 files, +682 -60)

| File | Change |
|------|--------|
| `ProfileScheduleTime.swift` | Add `scheduledStartTime`, `activeWindowStart`, `shouldBeActiveNow`; remove `shouldSuppressStart` |
| `ScheduleTimerActivity.swift` | V2 path uses `shouldBeActiveNow`, legacy retains individual guards |
| `PreActivationReminderScheduler.swift` | Add `catchUpMissedScheduleStarts` |
| `foqosApp.swift` | Wire catch-up into both foreground handlers |
| `BlockedProfiles.swift` | Rename property + snapshot |
| `Shared.swift` | Rename in ProfileSnapshot |
| `BlockedProfileSessions.swift` | Simplify writer: `= endTime` |
| `StrategyTimerActivity.swift` | Simplify writer: `= Date()` |
| `TriggerConfigurationModel.swift` | Remove suppression from `saveToProfile` |
| `SyncModels.swift` | Rename property + FieldKey |
| `SyncCoordinator.swift` | Rename in update + create paths |
| `cloudkit-schema.ckdb` | Rename field |
| `ActiveWindowTests.swift` | 9 tests for window detection |
| `ShouldBeActiveNowTests.swift` | 13 tests for consolidated helper |
| `ScheduleSuppressionTests.swift` | 8 tests for suppression via `shouldBeActiveNow` |
| `ProfileScheduleTimeTests.swift` | 2 tests for `scheduledStartTime` |

## Test Plan
- [x] 9 tests for `activeWindowStart` — same-day, overnight, start-only, boundary cases
- [x] 13 tests for `shouldBeActiveNow` — window, day, age, suppression, overnight combinations
- [x] 8 tests for suppression scenarios via `shouldBeActiveNow` — including overnight suppression
- [x] 2 tests for `scheduledStartTime(on:)` — correct time construction
- [x] All test dates pinned to fixed reference (2026-06-15) for determinism
- [x] Full test suite passes
- [x] Build succeeds

Closes #96

### CODE REVIEW

**Comment by mnbf9rca** [comment]
Re: general feedback from @sourcery-ai

**"Use fixed Calendar with known timezone for test stability"** — The tests already use a fixed reference date (2026-06-15) for determinism. The production code uses `Calendar.current`, so testing with a different calendar (e.g., UTC) would test different behavior than what runs on device. The fixed reference date addresses the day-boundary flakiness concern.

**"Dense logic in activeWindowStart/shouldBeActiveNow, add inline comments"** — Both methods already have comprehensive doc comments explaining each case (same-day, overnight, start-only). The methods are ~30 lines each with clear branching structure. Additional inline comments would add noise to already well-documented code.

File: FoqosTests/ShouldBeActiveNowTests.swift [hunk 1]
```diff
@@ -0,0 +1,188 @@
+// FoqosTests/ShouldBeActiveNowTests.swift
+import XCTest
+
+@testable import FamilyFoqos
+
+final class ShouldBeActiveNowTests: XCTestCase {
+  private let calendar = Calendar.current
+
+  // Fixed reference: Wednesday 2026-06-15 (weekday 4 = Wednesday)
```
[the remark this answers is not in the document]
Comment (mnbf9rca) [reply FoqosTests/ShouldBeActiveNowTests.swift]: Fixed. Updated all 3 test files to correctly say Monday (weekday 2). Also fixed variable names and related comments referencing Wednesday/Tuesday. (9349769)
